### ISSUE TITLE
Fails if return value is a numpy array

### ISSUE THREAD

**Issue by smutch** [google/python-fire#13]
If the return value of a wrapped function is a numpy array then Fire raises the following error:

``` python
File "/home/.../miniconda3/lib/python3.5/site-packages/fire/core.py", line 430, in _Fire
  if component == last_component and remaining_args == initial_args:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

The problem is that any equality check with a numpy array will be carried out element-wise.

I tried to implement a quick fix by checking `hasattr(component, 'tolist')`, however, I couldn't obviously find a robust solution to this issue.

**Comment by dbieber** [comment]
Thanks for raising this issue.
From the error message you posted, it looks like the reason numpy arrays fail here is that they override the __eq__ function. We may be able to fix this issue by changing "component == last_component" to "component is last_component", which will check if the two objects are actually the same object, rather than running the __eq__ function.

Does this seem right? Can you check if this resolves the issue for you?

**Comment by dbieber** [comment]
I checked, and this two-character fix seems to do the trick :).

Here's a small example:

```
import fire
import numpy as np
x = np.array([[1, 2, 3], [4, 5, 6]])
fire.Fire()
```

`example.py x`

numpy arrays have lots you do with them, which means the output of Fire is pretty extensive. You'll probably usually want to run something like `example.py x tolist` instead, which will output:
```
[1, 2, 3]
[4, 5, 6]
```

**Comment by smutch** [comment]
Great!  Thanks for the quick response. I also confirm this does the trick.

I did briefly consider this too, but I wasn't sure if it could be guaranteed that both `component` and `last_component` wouldn't ever be copies of one another.  Great to hear from someone who knows the code, and what they're doing, that this works. 😄 

Thanks for the tip with the `tolist` use on the command line as well.  I'm only just starting to get to grips with the power of this package, but I'm convinced I'll be using it a lot for my scripts...

**Comment by dbieber** [comment]
Resolved by 88f65747e25c20901483ed1d29d79392bd9b6fe3
